spinta/datasets/backends/sql/commands/read.py

```diff
@@ -105,10 +102,18 @@ def getall(
                     if isinstance(sel.prop.dtype, PrimaryKey):
                         val = keymap.encode(sel.prop.model.model_type(), val)
                     elif isinstance(sel.prop.dtype, Ref):
-                        val = keymap.encode(sel.prop.dtype.model.model_type(), val)
+                        if sel.prop.dtype.refprops == sel.prop.dtype.model.external.pkeys or len(
+                            sel.prop.dtype.model.external.pkeys) == 1:
+                            val = keymap.encode(sel.prop.dtype.model.model_type(), val)
+                        else:
+                            pkey = keymap.encode(sel.prop.model.model_type(),
+                                                 sel.prop.dtype.model.external.pkeys[-1].name)
+                            val = keymap.encode(
+                                f'{sel.prop.dtype.model.model_type()}.{sel.prop.dtype.model.external.pkeys[-1].name}',
```

Comment:
Instead of using only last pkey, we should use all keys. And I think we need a helper function here, that would generate a correct keymap name, from given model and optionally prop.dtype.refprops.